frontend/components/CookieBanner.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { Component } from 'react';
+import { css } from 'react-emotion';
+import { palette } from '@guardian/pasteup/palette';
+import { sans, serif } from '@guardian/pasteup/fonts';
+import { Container } from '@guardian/guui';
+import TickIcon from '@guardian/pasteup/icons/tick.svg';
+import { getCookie, addCookie } from '../lib/cookie';
+
+const banner = css`
+    position: fixed;
+    bottom: 0;
+    background-color: ${palette.sport.dark};
+    padding: 10px 0 24px;
+    width: 100%;
+
+    p {
+        font-family: ${serif.body};
+        font-size: 17;
+        line-height: 24px;
+        margin-top: 0;
+        margin-bottom: 8px;
+        color: ${palette.neutral[100]};
+    }
+
+    a {
+        color: ${palette.neutral[100]};
+        border-bottom: 1px solid rgba(255, 255, 255, 0.5);
+        text-decoration: none;
+
+        :hover {
+            border-color: ${palette.neutral[100]};
+        }
+    }
+`;
+
+const inner = css`
+    padding: 0 10px;
+`;
+
+const header = css`
+    font-size: 34px;
+    line-height: 38px;
+    font-weight: bold;
+    padding-bottom: 12px;
+    color: ${palette.highlight.main};
+`;
+
+const more = css`
+    font-size: 14px;
+    line-height: 20px;
+    font-family: ${sans.body};
+    margin-left: 12px;
+`;
+
+const p = css``;
+
+const button = css`
+    font-size: 16px;
+    line-height: 24px;
+    font-family: ${sans.body};
+    border-radius: 1000px;
+    height: 42px;
+    background: ${palette.highlight.main};
+    color: ${palette.neutral[7]};
+    padding: 0 25px 0 46px;
+    display: block;
+    align-items: flex-start;
+    justify-content: space-between;
+    cursor: pointer;
+    position: relative;
+    border: 0;
+    white-space: nowrap;
+    display: inline-block;
+
+    svg {
+        position: absolute;
+        top: 0;
+        left: 0;
+        bottom: 0;
+        margin: auto;
+        width: 42px;
+        height: 42px;
+        transform: scale(0.5);
+    }
+`;
+
+const actions = css`
+    margin-top: 24px;
+`;
+
+const consentCookie = 'GU_TK';
+
+export default class CookieBanner extends Component<{}, { show: boolean }> {
+    constructor(props: {}) {
+        super(props);
+
+        this.state = {
+            show: false,
+        };
+
+        this.accept = this.accept.bind(this);
```

Comment:
do typescript public methods need binding?

Reply:
https://basarat.gitbooks.io/typescript/docs/tips/bind.html and https://stackoverflow.com/questions/20627138/typescript-this-scoping-issue-when-called-in-jquery-callback
seem to suggest `public accept = () =>` would be better

Reply:
I'm pretty sure it didn't work without the binding but will check.

Reply:
Cool thanks - that does look nicer!